### PR TITLE
Updated docs for the new Write Access profile API

### PR DESCRIPTION
**Pull request by lemunozm**

### CODE REVIEW

File: docs/client.rst [hunk 1]
```diff
@@ -543,23 +546,30 @@ If there is an error, a status error will be sent by the agent.
 
 Write access profile
 ````````````````````
-These functions are generated automatically by `MicroRTPSGen` utility with the ``-write-access-profile`` option enabled over an idl file with a topic `TOPICTYPE`.
-The declaration of these function can be found in the generated file ``TOPICTYPEWriter.h``.
+These functions are enabled when PROFILE_WRITE_ACCESS is enabled into ``client.config`` file.
```
Comment (BorjaOuterelo) [inline docs/client.rst]: Use "in" instead of "into".
"Enabled in the ..."

File: docs/client.rst [hunk 2]
```diff
@@ -543,23 +546,30 @@ If there is an error, a status error will be sent by the agent.
...
 
-This function writes a topic into a stream.
-If the returned value is ``true``, the topic has been serialized.
+Requests a writing into a specific output stream.
+For that this function will initialize a ``MicroBuffer`` struct where a topic of ``topic_size`` size must be serialized.
+If the returned value is ``true``, exists the necessary gap for writing a ``topic_size`` bytes into the stream.
+If the returned value is ``false``, the topic can no be serialized into the stream.
 The topic will be sent in the next ``run_session`` function.
 
+NOTE: All `topic_size` bytes requested will be sent to the agent after a ``run_session`` call, no matter if the ``MicroBuffer`` has benn used or not.
```
Comment (BorjaOuterelo) [inline docs/client.rst]: "has been"

File: docs/client.rst [hunk 3]
```diff
@@ -543,23 +546,30 @@ If there is an error, a status error will be sent by the agent.
...
+
 :session: Session structure previously initialized.
 :stream_id: The output stream ID where the message will be written.
-:object_id: The DataWriter ID that will write the topic to the DDS World.
-:topic: The topic that will be sent to the agent.
+:datawriter_id: The DataWriter ID that will write the topic to the DDS World.
+:mb_topic: A ``MicroBuffer`` struct used to serialize the topic.
+           This struct points to a requested gap into the stream.
+:topic_size: The bytes that will be reserved into the stream.
```
Comment (BorjaOuterelo) [inline docs/client.rst]: "reserved in" instead of "reserve into"

File: docs/getting_started.rst [hunk 4]
```diff
@@ -209,14 +207,20 @@ For creating a message with data, first we must to decide which stream we want t
...
     mr_run_session_until_confirmed_delivery(&session, 1000);
 
-``mr_write_HelloWorld_topic`` function is automatically generated by :ref:`micrortpsgen_label` from the IDL.
-This function serializes the topic into stream.
-If the stream is available and the topic fix into it, a valid *request id* is returned.
-``datawriter_id`` correspond to the data writer entity used for sending the data.
+``HelloWorld_size_of_topic`` and ``HelloWorld_serialize_topic`` functions are automatically generated by :ref:`micrortpsgen_label` from the IDL.
+The function ``mr_prepare_output_stream`` request a writing for a topic of ``topic_size`` size into the reliable stream represented by ``reliable_out``,
+with a ``datawriter_id`` (correspond to the data writer entity used for sending the data in the `DDS World`).
+If the stream is available and the topic fit into it, the function will initialize the ``MicroBuffer`` structure ``mb``.
```
Comment (BorjaOuterelo) [inline docs/getting_started.rst]: "fit in it"

File: docs/getting_started.rst [hunk 5]
```diff
@@ -209,14 +207,20 @@ For creating a message with data, first we must to decide which stream we want t
...
+Once the ``MicroBuffer`` is prepared, the topic can be serialized into it.
+We have careless about ``mr_prepare_output_stream`` return value because the serialization only will occurs if the ``MicroBuffer` is valid``
```
Comment (BorjaOuterelo) [inline docs/getting_started.rst]: "We are careless" and "the serialization will only occur"

File: docs/client.rst [hunk 6]
```diff
@@ -603,6 +613,7 @@ It counts the number of bytes that the topic will need in a `MicroBuffer`.
 
 :topic: Struct to count the size.
 :size: Number of bytes already written into the `MicroBuffer`.
+       Typically its value is `0` if the purpose is to use into ``mr_prepare_output_stream`` function.
```
Comment (BorjaOuterelo) [inline docs/client.rst]: "to use in"

File: docs/client.rst (hunk 1 above)
Comment (lemunozm) [reply docs/client.rst]: Done!

File: docs/client.rst (hunk 2 above)
Comment (lemunozm) [reply docs/client.rst]: Done!

File: docs/client.rst (hunk 3 above)
Comment (lemunozm) [reply docs/client.rst]: Done!

File: docs/getting_started.rst (hunk 4 above)
Comment (lemunozm) [reply docs/getting_started.rst]: Done!

File: docs/getting_started.rst (hunk 5 above)
Comment (lemunozm) [reply docs/getting_started.rst]: done

File: docs/client.rst (hunk 6 above)
Comment (lemunozm) [reply docs/client.rst]: Done!

File: docs/getting_started.rst [hunk 7]
```diff
@@ -209,14 +207,20 @@ For creating a message with data, first we must to decide which stream we want t
...
+    (void) HelloWorld_serialize_topic(&mb, &topic);
 
     mr_run_session_until_confirmed_delivery(&session, 1000);
 
-``mr_write_HelloWorld_topic`` function is automatically generated by :ref:`micrortpsgen_label` from the IDL.
-This function serializes the topic into stream.
-If the stream is available and the topic fix into it, a valid *request id* is returned.
-``datawriter_id`` correspond to the data writer entity used for sending the data.
+``HelloWorld_size_of_topic`` and ``HelloWorld_serialize_topic`` functions are automatically generated by :ref:`micrortpsgen_label` from the IDL.
+The function ``mr_prepare_output_stream`` request a writing for a topic of ``topic_size`` size into the reliable stream represented by ``reliable_out``,
```
Comment (richiware) [inline docs/getting_started.rst]: It should be `requests`

File: docs/getting_started.rst [hunk 8]
```diff
@@ -209,14 +207,20 @@ For creating a message with data, first we must to decide which stream we want t
...
+with a ``datawriter_id`` (correspond to the data writer entity used for sending the data in the `DDS World`).
+If the stream is available and the topic fit in it, the function will initialize the ``MicroBuffer`` structure ``mb``.
```
Comment (richiware) [inline docs/getting_started.rst]: It should be `fits`

File: docs/getting_started.rst (hunk 7 above)
Comment (lemunozm) [reply docs/getting_started.rst]: done!

File: docs/getting_started.rst (hunk 8 above)
Comment (lemunozm) [reply docs/getting_started.rst]: done!

**Pull request merged**